lib/fetching/content-services/facebook-content-service.js

```diff
@@ -26,116 +27,139 @@ FacebookContentService.prototype._doFetch = function(options, callback) {
+    });
     callback([]);
     return;
   }
 
-  var queries = {};
+  var source = this._url.trim();
 
-  // Determine Source ID of object
-  queries.source = 'SELECT id, site, type, url FROM object_url WHERE url = "' + this._url + '"';
+  // Parse out source name from url
+  if (source.indexOf('?') === -1) {
+    source = source.split('?')[0];
+  }
 
-  // Fetch posts and comments
-  queries.posts = 'SELECT source_id, post_id, updated_time, created_time, actor_id, permalink, message';
-  queries.posts += ' FROM stream';
-  queries.posts += ' WHERE source_id IN (SELECT id FROM #source)';
-  // New comments trigger a fresh updated_time in its parent post
-  if (this._lastReportDate) queries.posts += ' AND updated_time > ' + (this._lastReportDate.getTime()/1000);
-  queries.posts += ' ORDER BY updated_time';
+  if (source.lastIndexOf('/') === source.length - 1) {
+    source = source.substring(0, source.length - 1);
+  }
 
-  // Get new comments associated with posts.
-  queries.comments = "SELECT id, fromid, time, text, post_id FROM comment WHERE (post_id IN (SELECT post_id FROM #posts))";
-  if (this._lastReportDate) queries.comments += ' AND time > ' + (this._lastReportDate.getTime()/1000);
-  queries.comments += ' ORDER BY time';
+  source = source.substring(source.lastIndexOf('/') + 1, source.length);
 
-  queries.authors = 'SELECT id, name FROM profile WHERE id IN (SELECT actor_id FROM #posts) OR id IN (SELECT fromid FROM #comments)';
 
-  this._doRequest(queries, function(err, res) {
+  this._doRequest(source, function(err, data) {
     if (err) {
-      self.emit('error', new Error(err.message))
+      self.emit('error', new Error(err.message));
       return callback([]);
     }
+    callback(self._handleResults(data, source));
+  })
+};
 
-    // This will be an array of info about the source.
-    var sourceInfo = _.findWhere(res.data, {name: 'source'}).fql_result_set;
+FacebookContentService.prototype._doRequest = function (source, callback) {
+  var self = this;
+  var data = [];
+  var params = {};
 
-    if (sourceInfo.length === 0) {
-      self.emit('error', new Error('Invalid Facebook URL'));
-      callback([]);
-    } else {
-      var posts = _.findWhere(res.data, {name: 'posts'}).fql_result_set;
-      var comments = _.findWhere(res.data, {name: 'comments'}).fql_result_set;
-      var authors = _.findWhere(res.data, {name: 'authors'}).fql_result_set;
-      callback(self._handleResults(posts, comments, authors));
-    }
-  });
+  if (this._lastReportDate) {
+    params.since = this._lastReportDate.getTime() / 1000;
+  }
+  // Grab posts off of sources feed
+  self._doGet('/v2.2/' + source + '/feed', params, data, callback);
+
+  //If no new posts, grab comments off of most recent post
+  if((!data || data.length == 0) && this.posts[source] != null) {
+    self._doGet('/v2.2/' + this.posts[source].id + '/comments?order=reverse_chronological', params, data, callback);
```

Comment:
I don't quite get how this is supposed to work. Why are you only loading comments if there no new posts? There could be both new posts and new comments on older posts, etc. There are also comment replies, which I think are not being handled here at all... The beauty of FQL was that it allowed grabbing all these things fairly easily I remember. But with the Graph API it might be trickier.